modules/core/shared/src/main/scala/gem/config/GnirsConfig.scala

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.config
+
+import cats.Eq
+import gem.enum.{ GnirsFpuSlit, GnirsFpuOther }
+
+/** Additional type hierarchy over the low-level GNIRS enums.
+  */
+object GnirsConfig {
+
+  /** Focal plane unit choice, slit or other. */
+  sealed trait GnirsFpu extends Product with Serializable {
+
+    import GnirsFpu.{ Slit, Other }
+
+    /** Extracts the FPU Slit, if applicable. */
+    def toFpuSlit: Option[GnirsFpuSlit] =
+      this match {
+        case Slit(fpu) => Some(fpu)
+        case _         => None
+      }
+
+    /** Extracts the FPU Other, if applicable. */
+    def toFpuOther: Option[GnirsFpuOther] =
+      this match {
+        case Other(fpu) => Some(fpu)
+        case _          => None
+      }
+
```

Comment:
A `fold` might come in handy?

```scala
def fold[A](slit: GnirsFpuSlit => A, other: GnirsFpuOther => A): A =
  this match {
    case Slit(fpu)  => slit(fpu)
    case Other(fpu) => other(fpu)
  }

def toFpuSlit: Option[GnirsFpuSlit] =
  fold(Option(_), None)

  ...
```

Or thinking about it a bit more, is there any reason this isn't just a type alias?

```scala
type GnirsFpu = Either[GnirsFpuOther, GnirsFpuSlit]
```

Reply:
In a review a while back I was told that, in Gemini code, the `Left` case of an `Either` was usually associated with an error or fallback connotation, so since then, I was always creating distinct sum types when there is no associated connotation for the `Left` case.

Speaking for myself, I never assume any connotation for `Left` anyway, only that, for whatever reason, there is slight bias towards `Right`, so I'm fine with making this an `Either`.

Reply:
Hmm, if anybody holds the opinion that `Either` is only for errors we should discuss it.  There may be good reasons for that, but if so I'm ignorant of them.  At any rate, when I had to make a similar call for GMOS FPU I went with `Either[GmosCustomMask, GmosNorthFpu]`.